chain/network/src/peer_manager/connection_store/mod.rs

```diff
@@ -87,28 +68,77 @@ impl Inner {
     }
 }
 
-pub(crate) struct ConnectionStore(Mutex<Inner>);
+pub(crate) struct ConnectionStore(ArcMutex<Inner>);
 
 impl ConnectionStore {
-    pub fn new(store: store::Store) -> anyhow::Result<Self> {
+    pub(crate) fn new(store: store::Store) -> anyhow::Result<Self> {
```

Comment:
when choosing visibility for methods, consider whether they are supposed to be a part of their API (then use just pub) or an implementation detail (then private). Do not use pub(crate) only because the ConnectionStore is pub(crate). If ConnectionStore was pub, then these methods would have to be pub as well, right?